library/common/main_interface.cc

```diff
@@ -157,6 +159,64 @@ envoy_status_t record_histogram_value(envoy_engine_t, const char* elements, envo
   return ENVOY_FAILURE;
 }
 
+namespace {
+struct AdminCallContext {
+  envoy_status_t status_{};
+  envoy_data response_{};
+
+  absl::Mutex mutex_{};
+  absl::Notification data_received_{};
+};
+
+absl::optional<envoy_data> blockingAdminCall(absl::string_view path, absl::string_view method,
```

Comment:
maybe a test that verifies that `POST` calls work too - not a blocker but would be nice to verify it (since POST is not used at all as of now). Or just hardcode it to `GET` 🤷